### PR TITLE
Feat/crm 42 edit company info BE

### PR DESCRIPTION
**Pull request by charith-rc**
### Summary

-Adds an edit companies enpoint

### How to test

1.

### Project Checklist

- [x] Changes build without any errors
- [ ] Have written adequate test cases
- [x] Done developer testing in
  - [ ] Chrome
  - [ ] Firefox
  - [ ] Safari
- [x] Code is formatted with `npm run format`
- [x] Code is linted with `npm run check-lint`
- [x] No unnecessary comments left in code
- [ ] Made corresponding changes to the documentation

#### Other

- [ ] New atomic components added
- [ ] New molecules added
- [ ] New pages(routes) added
- [ ] New dependencies installed

### PR Checklist

- [x] Pull request is raised from the correct source branch
- [x] Pull request is raised to the correct destination branch
- [x] Pull request is raised with correct title
- [x] Pull request is self reviewed
- [x] Pull request is self assigned
- [ ] Suitable pull request status labels are added (`ready-for-code-review`)

### Additional Information
-

### CODE REVIEW

File: backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java [hunk 1]
```diff
@@ -143,4 +143,37 @@ public ResponseEntityDto deleteCompany(Long id) {
...
+		if (!existingCompany.getName().equalsIgnoreCase(crmCompany.getName())
+				&& checkCompanyExists(crmCompany.getName())) {
+			throw new ModuleException(CrmMessageConstant.CRM_ERROR_COMPANY_EXISTS);
+		}
+
+		existingCompany.setName(crmCompany.getName());
+		existingCompany.setContactNumber(crmCompany.getContactNumber());
+		existingCompany.setWebsite(crmCompany.getWebsite());
+		existingCompany.setAddress(crmCompany.getAddress());
+		existingCompany.setIndustry(crmCompany.getIndustry());
```
[the remark this answers is not in the document]
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java]: frontend implementation would autofill form with existing details as per requirement, thus modelling this as a PUT seemed appropriate. required fields (name) enforced in validation.

File: backend/src/main/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyController.java [hunk 2]
```diff
@@ -76,4 +77,13 @@ public ResponseEntity<ResponseEntityDto> deleteCompany(@PathVariable Long id) {
 		return new ResponseEntity<>(responseDto, HttpStatus.OK);
 	}
 
+	@Operation(summary = "Edit a company by ID", description = "Edit information related to a registered company by ID")
+	@PutMapping("/{id}")
+	@PreAuthorize("hasAnyRole('ROLE_CRM_SALES_MANAGER')")
+	public ResponseEntity<ResponseEntityDto> editCompany(@PathVariable Long id,
+			@RequestBody CrmCompanyCreateDto crmCompany) {
+		ResponseEntityDto responseDto = companyService.editCompany(id, crmCompany);
```
[the remark this answers is not in the document]
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyController.java]: added integration tests 4ee2d21eb9c791f1ddf309533de0e9be84a64ad2

**Review by ThinuwanW — COMMENTED** [review]
## 🤖 Claude Code Review

Adds a PUT /v1/crm/company/{id} endpoint with manager-only authorization, mirroring the create flow's validation and duplicate-name check while skipping the check when the name is unchanged. The implementation is functionally correct and well-tested for the main paths, but the PUT-with-null-fields semantics, some missing edge-case tests, and a wider-impact data.sql change deserve attention.

Found **17** new issue(s): 🔴 1 important, 🟡 7 suggestion(s), 🟣 9 nit(s)

File: backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java (hunk 1 above)
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java]: 🤖 **Claude** · 🔴 **Important**

Partial PUT payloads silently wipe optional fields. The validators for contactNumber/website/address/industry early-return on null/blank, and the edit method unconditionally calls setContactNumber/setWebsite/setAddress/setIndustry with whatever was in the DTO. A client that submits {"name":"X"} (or omits any of these fields) will null out the previously-stored values without any error. Either (a) document this as strict PUT-replace semantics, (b) only copy non-null fields (PATCH-like) — e.g. `if (crmCompany.getWebsite() != null) existingCompany.setWebsite(...)`, or (c) require all fields in the DTO via @NotNull / explicit validation. Without one of these, frontend bugs will corrupt company records.

File: backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java [hunk 3]
```diff
@@ -143,4 +143,37 @@ public ResponseEntityDto deleteCompany(Long id) {
...
+
+		CrmCompany existingCompany = crmCompanyDao.findByIdAndIsDeletedFalse(id)
+			.orElseThrow(() -> new ModuleException(CrmMessageConstant.CRM_ERROR_COMPANY_NOT_FOUND));
+
+		if (!existingCompany.getName().equalsIgnoreCase(crmCompany.getName())
+				&& checkCompanyExists(crmCompany.getName())) {
+			throw new ModuleException(CrmMessageConstant.CRM_ERROR_COMPANY_EXISTS);
+		}
+
+		existingCompany.setName(crmCompany.getName());
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java]: 🤖 **Claude** · 🟡 **Suggestion**

`crmCompany.getName()` is set onto the entity verbatim (only the `name` field has TrimmingStringDeserializer; the other optional fields do not). Consider whether contactNumber/address/industry/website should also be trimmed or normalized on edit (and on create) for consistency — at minimum the inconsistency with `name` is worth a comment.

File: backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java [hunk 4]
```diff
@@ -143,4 +143,37 @@ public ResponseEntityDto deleteCompany(Long id) {
...
+		existingCompany.setContactNumber(crmCompany.getContactNumber());
+		existingCompany.setWebsite(crmCompany.getWebsite());
+		existingCompany.setAddress(crmCompany.getAddress());
+		existingCompany.setIndustry(crmCompany.getIndustry());
+
+		CrmCompany updatedCompany = crmCompanyDao.save(existingCompany);
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java:186]: 🤖 **Claude** · 🟣 **Nit**

`crmCompanyDao.save(existingCompany)` is redundant: `existingCompany` is a managed JPA entity inside an @Transactional method, so dirty checking will flush automatically. Harmless, but matches an unnecessary pattern; could be removed for clarity, or kept for symmetry with createCompany — your call.

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java [hunk 5]
```diff
@@ -268,4 +279,79 @@ void deleteCompany_WithoutManagerRole_ReturnsForbidden() throws Exception {
...
+	}
+
+	@Test
+	@DisplayName("Edit company without CRM manager role - Returns Forbidden")
+	void editCompany_WithoutManagerRole_ReturnsForbidden() throws Exception {
+		authToken = jwtService.generateAccessToken(userDetailsService.loadUserByUsername("user2@gmail.com"), 1L);
+
+		performPutRequest(1L, createValidPayload()).andDo(print()).andExpect(status().isForbidden());
+	}
+
```
Comment (ThinuwanW) [inline backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java:377]: 🤖 **Claude** · 🟡 **Suggestion**

Test coverage gaps for the new edit endpoint: (1) no test for editing a company while keeping the same name (verifies the !equalsIgnoreCase short-circuit skips the duplicate check on the entity itself), (2) no test for editing with a case-only change to the same name (e.g. "Acme Corp" → "ACME CORP"), and (3) no tests for input-validation failures (blank name, too-long name/address/industry/website, invalid contactNumber, malformed website). Since the validators are shared with create, the risk is lower, but explicit edit-side tests will catch regressions if validators diverge later.

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java [hunk 6]
```diff
@@ -58,6 +60,8 @@ class CrmCompanyControllerIntegrationTest {
 
 	private static final String DELETE_PATH = BASE_PATH + "/{id}";
 
+	private static final String EDIT_PATH = BASE_PATH + "/{id}";
```
Comment (ThinuwanW) [inline backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java:67]: 🤖 **Claude** · 🟣 **Nit**

`EDIT_PATH` is defined as the exact same string as `DELETE_PATH` (`BASE_PATH + "/{id}"`). Consider consolidating into a single constant (e.g. `ITEM_PATH` or `ID_PATH`) to avoid the duplication and make the intent — "any single-resource path" — clearer.

File: backend/src/test/resources/data.sql [hunk 7]
```diff
@@ -61,7 +61,7 @@ VALUES (default, 1, 1, 1, 'PRIMARY'),
 
 INSERT INTO employee_role (employee_id, people_role, leave_role, attendance_role, crm_role, is_super_admin)
 VALUES (1, 'PEOPLE_ADMIN', 'LEAVE_ADMIN', 'ATTENDANCE_ADMIN', 'CRM_ADMIN', true),
-       (2, 'PEOPLE_ADMIN', 'LEAVE_ADMIN', 'ATTENDANCE_ADMIN', null, false),
+       (2, 'PEOPLE_ADMIN', 'LEAVE_ADMIN', 'ATTENDANCE_ADMIN', 'CRM_SALES_REPRESENTATIVE', false),
```
Comment (ThinuwanW) [inline backend/src/test/resources/data.sql]: 🤖 **Claude** · 🟡 **Suggestion**

Changing user2's `crm_role` from null to `CRM_SALES_REPRESENTATIVE` is a global seed-data change that affects every integration test, not just CRM tests. The existing `deleteCompany_WithoutManagerRole_ReturnsForbidden` and new `editCompany_WithoutManagerRole_ReturnsForbidden` still pass (sales rep ≠ manager), and arguably this is a more realistic "no manager role" test than "no CRM role at all". But: any future test (or existing test in another module) that assumed user2 has no CRM authorities can now incorrectly succeed against CRM_SALES_REPRESENTATIVE-protected endpoints (e.g. /v1/crm/company/lookup, /exists, POST /company, etc.). Worth calling out in the PR description so other contributors know.

File: backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java [hunk 8]
```diff
@@ -143,4 +143,37 @@ public ResponseEntityDto deleteCompany(Long id) {
...
+		CrmValidations.validateCompanyName(crmCompany.getName());
+		CrmValidations.validateContactNumber(crmCompany.getContactNumber());
+		CrmValidations.validateWebsite(crmCompany.getWebsite());
+		CrmValidations.validateAddress(crmCompany.getAddress());
+		CrmValidations.validateIndustry(crmCompany.getIndustry());
+
+		CrmCompany existingCompany = crmCompanyDao.findByIdAndIsDeletedFalse(id)
+			.orElseThrow(() -> new ModuleException(CrmMessageConstant.CRM_ERROR_COMPANY_NOT_FOUND));
+
+		if (!existingCompany.getName().equalsIgnoreCase(crmCompany.getName())
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java]: 🤖 **Claude** · 🟣 **Nit**

No optimistic locking (no @Version on CrmCompany), so concurrent edits by two managers will last-write-wins silently overwrite each other. Same risk applies to other CRM entities, so this is not introduced by this PR — but the new edit endpoint is the first place where lost-update on a CrmCompany becomes user-visible. Consider whether @Version should be added now while you're touching the update path.

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java [hunk 9]
```diff
@@ -249,6 +259,7 @@ void deleteCompany_WithAssociatedDeals_SoftDeletesDeals() throws Exception {
 		deal.setCompany(crmCompanyDao.getReferenceById(companyId));
 		deal.setContact(contact);
 		deal.setOwner(employeeDao.getReferenceById(1L));
+		deal.setPriority(CrmDealPriority.MEDIUM);
```
Comment (ThinuwanW) [inline backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java:276]: 🤖 **Claude** · 🟡 **Suggestion**

Scope creep: adding `deal.setPriority(CrmDealPriority.MEDIUM)` is unrelated to the edit-company feature — it fixes a latent NOT NULL constraint issue in the existing `deleteCompany_WithAssociatedDeals_SoftDeletesDeals` test. Bundling it with the edit-company change hides the fix from reviewers and complicates a potential revert. Move this (plus the `CrmDealPriority` import on line 10) to a separate commit/PR so the test fix has its own review trail.

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java [hunk 10]
```diff
@@ -268,4 +279,79 @@ void deleteCompany_WithoutManagerRole_ReturnsForbidden() throws Exception {
 		performDeleteRequest(1L).andDo(print()).andExpect(status().isForbidden());
 	}
 
+	// --- Edit company tests ---
```
Comment (ThinuwanW) [inline backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java:296]: 🤖 **Claude** · 🟡 **Suggestion**

Missing validation-failure tests for the edit endpoint. There is an analog of `createCompany_BlankName_ReturnsBadRequest` for create, but none of the validators (blank name, name > 30 chars, invalid website URL, contact number > 15 chars, etc.) are exercised against PUT /v1/crm/company/{id}. These code paths in `CrmValidations.validateCompanyName/Website/etc.` are now reachable from a second endpoint but only tested via create — add at least one negative test (e.g. blank name returns Bad Request) for the edit path.

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java [hunk 11]
```diff
@@ -268,4 +279,79 @@ void deleteCompany_WithoutManagerRole_ReturnsForbidden() throws Exception {
...
+			.andExpect(status().isOk())
+			.andExpect(jsonPath(STATUS_PATH).value(STATUS_SUCCESSFUL))
+			.andExpect(jsonPath(RESULTS_0_PATH + "['name']").value("Acme Corp Updated"))
+			.andExpect(jsonPath(RESULTS_0_PATH + "['industry']").value("Finance"))
+			.andExpect(jsonPath(RESULTS_0_PATH + "['website']").value("https://acme-updated.com"))
+			.andExpect(jsonPath(RESULTS_0_PATH + "['address']").value("456 New St"))
+			.andExpect(jsonPath(RESULTS_0_PATH + "['contactNumber']").value("94779876543"));
+	}
+
+	@Test
```
Comment (ThinuwanW) [inline backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java:332]: 🤖 **Claude** · 🟡 **Suggestion**

Missing test: editing a soft-deleted company. `findByIdAndIsDeletedFalse` is the only thing protecting the edit path from resurrecting a soft-deleted company, but no test asserts that posting a PUT to a soft-deleted ID returns CRM_ERROR_COMPANY_NOT_FOUND. Add a test that deletes a company (similar to `deleteCompany_AlreadyDeleted_ReturnsBadRequest`) and then attempts to edit it.

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java [hunk 12]
```diff
@@ -268,4 +279,79 @@ void deleteCompany_WithoutManagerRole_ReturnsForbidden() throws Exception {
...
+			.asLong();
+
+		CrmCompanyCreateDto editDto = new CrmCompanyCreateDto();
+		editDto.setName("Acme Corp Updated");
+		editDto.setIndustry("Finance");
+		editDto.setWebsite("https://acme-updated.com");
+		editDto.setAddress("456 New St");
+		editDto.setContactNumber("94779876543");
+
+		performPutRequest(companyId, editDto).andDo(print())
```
Comment (ThinuwanW) [inline backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java]: 🤖 **Claude** · 🟡 **Suggestion**

The happy-path test only asserts the HTTP response body. It does not verify that the changes were actually persisted — a buggy controller that echoed the input DTO without saving would still pass. Add a follow-up GET (or a `crmCompanyDao.findById(...)` assertion) to confirm the entity in the DB reflects the updated fields.

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java [hunk 13]
```diff
@@ -268,4 +279,79 @@ void deleteCompany_WithoutManagerRole_ReturnsForbidden() throws Exception {
...
+		CrmCompanyCreateDto editDto = createValidPayload();
+
+		performPutRequest(Long.MAX_VALUE, editDto).andDo(print())
+			.andExpect(status().isBadRequest())
+			.andExpect(jsonPath(STATUS_PATH).value(STATUS_UNSUCCESSFUL))
+			.andExpect(jsonPath(RESULTS_0_PATH + MESSAGE_PATH)
+				.value(messageUtil.getMessage(CrmMessageConstant.CRM_ERROR_COMPANY_NOT_FOUND)));
+	}
+
+	@Test
```
Comment (ThinuwanW) [inline backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java:344]: 🤖 **Claude** · 🟣 **Nit**

`editCompany_DuplicateNameOfAnotherCompany_ReturnsBadRequest` uses identical casing for both names (`Acme Corp` → `Acme Corp`), so it never exercises the `equalsIgnoreCase` / `existsByNameIgnoreCaseAndIsDeletedFalse` case-insensitivity branch — a straight equals would pass this test. Vary one side (e.g. edit to `ACME CORP`) to actually cover the case-insensitive duplicate-detection logic introduced on line 161.

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java [hunk 14]
```diff
@@ -268,4 +279,79 @@ void deleteCompany_WithoutManagerRole_ReturnsForbidden() throws Exception {
...
+			.andExpect(jsonPath(RESULTS_0_PATH + "['address']").value("456 New St"))
+			.andExpect(jsonPath(RESULTS_0_PATH + "['contactNumber']").value("94779876543"));
+	}
+
+	@Test
+	@DisplayName("Edit company with non-existent ID - Returns Bad Request")
+	void editCompany_NonExistentId_ReturnsBadRequest() throws Exception {
+		CrmCompanyCreateDto editDto = createValidPayload();
+
+		performPutRequest(Long.MAX_VALUE, editDto).andDo(print())
```
Comment (ThinuwanW) [inline backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java]: 🤖 **Claude** · 🟣 **Nit**

Using `Long.MAX_VALUE` as a not-found sentinel is brittle — any driver/dialect quirk around overflow or sequence interactions could produce a different failure mode than a normal non-existent ID. Prefer a plain unused value like `999_999L` to mirror real-world miss scenarios.

File: backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java [hunk 15]
```diff
@@ -143,4 +143,37 @@ public ResponseEntityDto deleteCompany(Long id) {
 		return new ResponseEntityDto(messageUtil.getMessage(CrmMessageConstant.CRM_SUCCESS_COMPANY_DELETED), false);
 	}
 
+	@Override
+	@Transactional
+	public ResponseEntityDto editCompany(Long id, CrmCompanyCreateDto crmCompany) {
+		log.info("editCompany: execution started");
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java:150]: 🤖 **Claude** · 🟣 **Nit**

`log.info("editCompany: execution started")` (and the matching end log) does not include the company ID being edited. The existing `deleteCompany` has the same omission, but since this is a new method being added, include `id` (and ideally the new name) in the log to make production debugging tractable, e.g. `log.info("editCompany: id={} new name={}", id, crmCompany.getName())`.

File: backend/src/test/resources/data.sql (hunk 7 above)
Comment (ThinuwanW) [inline backend/src/test/resources/data.sql]: 🤖 **Claude** · 🟣 **Nit**

Separate from the global-impact concern already raised: this seed change is not actually required by either of the new forbidden-role tests. `editCompany_WithoutManagerRole_ReturnsForbidden` (and the pre-existing delete equivalent) only need a user that lacks `ROLE_CRM_SALES_MANAGER`; user2 with `crm_role = null` satisfied that already. Reverting this line keeps the diff smaller and avoids changing user2's role for unrelated tests.

File: backend/src/main/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyController.java [hunk 16]
```diff
@@ -76,4 +77,13 @@ public ResponseEntity<ResponseEntityDto> deleteCompany(@PathVariable Long id) {
 		return new ResponseEntity<>(responseDto, HttpStatus.OK);
 	}
 
+	@Operation(summary = "Edit a company by ID", description = "Edit information related to a registered company by ID")
+	@PutMapping("/{id}")
+	@PreAuthorize("hasAnyRole('ROLE_CRM_SALES_MANAGER')")
+	public ResponseEntity<ResponseEntityDto> editCompany(@PathVariable Long id,
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyController.java:85]: 🤖 **Claude** · 🟣 **Nit**

Reusing `CrmCompanyCreateDto` for an edit endpoint couples the two operations' request shapes/validation forever. As soon as create or edit needs different rules (e.g. edit allowing an explicit null to clear a field, or create requiring fields edit treats as optional), this DTO has to be split anyway. Consider introducing `CrmCompanyEditDto` now while the surface area is small.

File: backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java [hunk 17]
```diff
@@ -143,4 +143,37 @@ public ResponseEntityDto deleteCompany(Long id) {
...
+		existingCompany.setContactNumber(crmCompany.getContactNumber());
+		existingCompany.setWebsite(crmCompany.getWebsite());
+		existingCompany.setAddress(crmCompany.getAddress());
+		existingCompany.setIndustry(crmCompany.getIndustry());
+
+		CrmCompany updatedCompany = crmCompanyDao.save(existingCompany);
+		CrmCompanyResponseDto responseDto = crmCompanyMapper.crmCompanyToCrmCompanyResponseDto(updatedCompany);
+
+		log.info("editCompany: execution ended successfully");
+		return new ResponseEntityDto(false, responseDto);
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java:190]: 🤖 **Claude** · 🟣 **Nit**

Inconsistent response shape with the sibling mutating endpoint: `deleteCompany` returns `new ResponseEntityDto(messageUtil.getMessage(...), false)` (a message DTO), while `editCompany` returns `new ResponseEntityDto(false, responseDto)` (the entity DTO with no success message). Either include a success message on edit for symmetry, or align both endpoints on one convention so API consumers don't have to special-case the response shape per verb.

File: backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java (hunk 1 above)
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java]: validations already cover required name field, others can be cleared as per requirement. as mentioned, frontend behaviour should be to auto fill existing values. a full payload is expected.

File: backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java (hunk 3 above)
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java]: name is the only unique field. since other fields are optional and only read in the detail view, trimmer was not added to keep the DTO clear

File: backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java (hunk 4 above)
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java:186]: keeping for consistency

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java (hunk 5 above)
Comment (charith-rc) [reply backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java:377]: only most important integration tests added

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java (hunk 6 above)
Comment (charith-rc) [reply backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java:67]: using a single constant would create coupling, may break with any future changes

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java (hunk 9 above)
Comment (charith-rc) [reply backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java:276]: added as a minor fix to failing test from deals dao update

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java (hunk 11 above)
Comment (charith-rc) [reply backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java:332]: guarded by jpa query anyway. omitted to prevent bulking integration tests

File: backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java (hunk 8 above)
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java]: keeping as existing pattern. Version annotation is not used.

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java (hunk 10 above)
Comment (charith-rc) [reply backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java:296]: validation unit tests seperately written. omitting to prevent bulkin integration tests.

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java (hunk 14 above)
Comment (charith-rc) [reply backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java]: wouldn't this break if test data ever gets seeded to this level? MAX_VALUE is guaranteed to be not found.

File: backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java (hunk 15 above)
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java:150]: no heavy logs

File: backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java (hunk 17 above)
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java:190]: existing pattern

File: backend/src/main/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyController.java (hunk 16 above)
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyController.java:85]: decoupled f59219308bd91c1187a182d2f1c37315081194bf

File: backend/src/test/resources/data.sql (hunk 7 above)
Comment (charith-rc) [reply backend/src/test/resources/data.sql]: reverted f59219308bd91c1187a182d2f1c37315081194bf

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java (hunk 13 above)
Comment (charith-rc) [reply backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java:344]: edited f59219308bd91c1187a182d2f1c37315081194bf

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java (hunk 12 above)
Comment (charith-rc) [reply backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java]: added f59219308bd91c1187a182d2f1c37315081194bf

File: backend/src/test/resources/data.sql (hunk 7 above)
Comment (charith-rc) [reply backend/src/test/resources/data.sql]: reverted f59219308bd91c1187a182d2f1c37315081194bf

**Review by akilarootcode — DISMISSED** [review]

**Review by akilarootcode — APPROVED** [review]

**Review by ThinuwanW — APPROVED** [review]
